ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/DecodePubsubMessages.java

```diff
@@ -1,56 +1,40 @@
 package com.mozilla.telemetry.transforms;
 
 import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
 
 /**
- * Base class for decoding PubsubMessage from various input formats.
- *
- * <p>Subclasses are provided via static members fromText() and fromJson().
- * We also provide an alreadyDecoded() transform for creating a PCollectionTuple
- * analogous to fromText or fromJson to handle output from PubsubIO.
- *
- * <p>The packaging of subclasses here follows the style guidelines as captured in
- * https://beam.apache.org/contribute/ptransform-style-guide/#packaging-a-family-of-transforms
+ * Container for transforms that decode PubsubMessage from various input formats.
  */
-public abstract class DecodePubsubMessages
-    extends MapElementsWithErrors.ToPubsubMessageFrom<String> {
-
-  // Force use of static factory methods.
-  private DecodePubsubMessages() {
-  }
+public class DecodePubsubMessages {
 
   /*
    * Static factory methods.
    */
 
   /** Decoder from non-json text to PubsubMessage. */
-  public static Text text() {
-    return new Text();
+  public static PTransform<PCollection<? extends String>, PCollection<PubsubMessage>> text() {
+    return PTransform.compose("DecodePubsubMessages.Text",
```

Comment:
Using PTransform.compose is a trick to be able to return a transform with a descriptive name rather than returning a transform with the generic name "Map". This trick doesn't work for LimitPayloadSize (where @scholtzan recently did a similar refactor) since that transform has to return a `Result` class.